src/components/PurchaseOrder/PO.js

```diff
@@ -36,28 +39,38 @@ class PO extends Component {
 
   static getDerivedStateFromProps(props, state) {
     const { parentMutator, parentResources } = props;
-    const initialValues = _.get(props, ['resources', 'order', 'records', 0]);
+    const initialValues = get(props, ['resources', 'order', 'records', 0]);
+
     // Set initialValues
     if (initialValues) {
-      if (!_.isEqual(initialValues, state.initialValues)) {
+      if (!isEqual(initialValues, state.initialValues)) {
         return { initialValues };
       }
     }
     // Check if initialValues STATE before updating child;
-    if (!_.isEmpty(state.initialValues)) {
+    if (!isEmpty(state.initialValues)) {
+      const createdBy = (parentResources.createdBy || {}).records || [];
       const vendor = (parentResources.vendor || {}).records || [];
       const user = (parentResources.user || {}).records || [];
-      if ((vendor || vendor.length > 0) || (user || user.length > 0)) {
+      const isAnyDataLoaded = vendor.length > 0 || user.length > 0 || createdBy.length > 0;
+
+      if (isAnyDataLoaded) {
         const initData = state.initialValues;
         const vendorName = vendor[0] && vendor[0].name ? `${vendor[0].name}` : '';
         const assignToName = user[0] && user[0].personal ? `${user[0].personal.firstName} ${user[0].personal.lastName}` : '';
-        if (vendorName !== initData.vendor_name || assignToName !== initData.assigned_to_user) {
+        const createdByName = createdBy[0] && createdBy[0].personal ? `${createdBy[0].personal.firstName} ${createdBy[0].personal.lastName}` : '';
```

Comment:
Please, move this `createdBy[0] && createdBy[0].personal` to a variable. Also is is possible to replace it with lodash `get` method. For example, get(createdBy, '0.personal')